x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/FieldAttribute.java

```diff
@@ -145,22 +144,53 @@ public static FieldAttribute readFrom(StreamInput in) throws IOException {
         return ((PlanStreamInput) in).readAttributeWithCache(FieldAttribute::new);
     }
 
+    private void writeParentName(StreamOutput out) throws IOException {
+        if (out.getTransportVersion().onOrAfter(TransportVersions.ESQL_FIELD_ATTRIBUTE_PARENT_SIMPLIFIED)) {
+            out.writeOptionalString(parentName);
+        } else {
+            if (parentName() == null) {
+                out.writeOptionalWriteable((FieldAttribute) null);
```

Comment:
This could just be `writeBoolean(false);`, but I guess that kind of exposes (and relies on) the internals of StreamInput/Output.

Reply:
That's true. I'd like to mirror the read and write methods as much as possible, though, to make the serialization code as obviously correct as possible - so I'd prefer to keep it this way.

Reply:
That's true. I'd like to mirror the read and write methods as much as possible, though, to make the serialization code as obviously correct as possible - so I'd prefer to keep it this way.

Reply:
That's true. I'd like to mirror the read and write methods as much as possible, though, to make the serialization code as obviously correct as possible - so I'd prefer to keep it this way.